src/scalib_ext/scalib/src/sasca/factor_graph.rs

```diff
@@ -322,18 +324,91 @@ impl FactorGraph {
                         ));
                     }
                 }
-                FactorKind::GenFactor { .. } => {
-                    // We cannot verify anything, except if we were given the
-                    // value of the generalized factors as argument of this
-                    // function (then we could check that the associated
-                    // probability is non-zero).
-                    // Let us not do any verification for know (code block
-                    // intentionally left empty).
+                FactorKind::GenFactor { id, operands } => {
+                    let ops: Vec<&PublicValue> = operands
+                        .iter()
+                        .map(|op| match op {
+                            GenFactorOperand::Var(idx, ..) => &var_assignments[*idx],
+                            GenFactorOperand::Pub(idx) => &public_values[*idx],
+                        })
+                        .collect();
+                    let mut ops_iter = ops.iter();
+                    let nmulti = loop {
+                        if let Some(PublicValue::Multi(x)) = ops_iter.next() {
+                            break x.len();
+                        }
+                    };
```

Comment:
This can get into an infinite loop if all ops are `Single` (here I set `numlti=1` in this case).

```suggestion
                    let nmulti_ops = ops.iter().filter_map(|op| if let PublicValue::Multi(x) { Some(x.len()) } else { None }).next();
                    let nmulti_factor = if let super::GenFactor::Multi(gfv) = &gen_factors[*id] { Some(gfv.len) } else { None };
                    let nmulti = match (numlti_ops, nmulti_factor) {
                        (Some(nm_ops), Some(nm_factors) if nm_ops == nm_factors => nm_factors,
                        (Some(nm_ops), Some(nm_factors) => todo!("Return an error because of mismatch"),
                        (Some(nmulti), None) | (None, Some(nmulti)) => nmulti,
                        (None, None) => 1
                    };
```